src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -404,7 +404,13 @@ public ReadQuery getQuery(QueryOptions options,
         boolean isPartitionRangeQuery = restrictions.isKeyRange() || restrictions.usesSecondaryIndexing();
 
         if (isPartitionRangeQuery)
+        {
+            if (restrictions.usesSecondaryIndexing()
+                && !SchemaConstants.isLocalSystemKeyspace(table.keyspace))
```

Comment:
super nit: I'd maybe make the condition a one-liner so it isn't aligned w/ the `Guardrails...` below :D